src/util/ChartUtils.ts

```diff
@@ -743,13 +742,13 @@ const STACK_OFFSET_MAP: Record<string, any> = {
   none: stackOffsetNone,
   silhouette: stackOffsetSilhouette,
   wiggle: stackOffsetWiggle,
-};
+} as const;
 
-export const getStackedData = (data: any, stackItems: any, offsetType: string) => {
+export const getStackedData = (data: any, stackItems: any, offsetType: keyof typeof STACK_OFFSET_MAP) => {
   const dataKeys = stackItems.map((item: any) => item.props.dataKey);
   const stack = shapeStack()
     .keys(dataKeys)
-    .value((d, key) => Math.max(+getValueByDataKey(d, key, 0), 0))
+    .value((d, key) => getValueByDataKey(d, key, 0))
```

Comment:
while this indeed fixes my issues it reverts what was done in https://github.com/recharts/recharts/commit/e458cb59043f79626ff60d1ecf033c0562876a24#diff-3943890095cad7d9dc4dd87a63a1c157 to fix another issue. Will have to check why that is.

Perhaps @xile611 can enlighten me :sweat_smile:
I saw this change was implemented to fix https://github.com/recharts/recharts/issues/1667 but I don't actually understand the bug :thinking: Why shouldn't negative values be negative? How would you recommend me to move forward with this? I guess I could special case this to `offsetType === 'sign' ? getValueByDataKey(d, key, 0)) : Math.max(+getValueByDataKey(d, key, 0), 0)` - would this still solve he issue you were trying to solve?

Reply:
I was thinking exactly the same to add this offsetType check before to do the Math.max, should we do that? I'm not an expert in d3 what does diverging do will it be the same?

Reply:
Thanks, I'll add a new kind of `offsetType = 'positive'` to resolve #1667 . And revert [e458cb5](https://github.com/recharts/recharts/commit/e458cb59043f79626ff60d1ecf033c0562876a24#diff-3943890095cad7d9dc4dd87a63a1c157)

Reply:
Thanks a million, just tested on beta.8 and it works as expected.